### PR TITLE
Change action to work and add UV+Solar Radiation

### PR DESCRIPTION
**Pull request by jschwerdlin**
Action would previously cause an error.  Also added UV and Solar Radiation upload.  Fixed warning about round() call not having a default.  Also fixed some issues with not uploading certain piece of data

### CODE REVIEW

File: wunderground_upload.yaml [hunk 1]
```diff
@@ -190,23 +204,23 @@ variables:
     {{ sensor_humidity if sensor_humidity == '' else (states(sensor_humidity) | float(default=0)) }}
   windspeedmph: >
     {% set unit_of_measurement = state_attr(sensor_windspeed, 'unit_of_measurement') %}
-    {{ sensor_windspeed if sensor_windspeed == '' or unit_of_measurement != 'km/h'
+    {{ (states(sensor_windspeed) | float(default=0)) if sensor_windspeed == '' or unit_of_measurement != 'km/h'
```
Comment (dvandonkelaar) [inline wunderground_upload.yaml:207]: Why is this change needed? The current code works, just like other sensors (for example humidity)

File: wunderground_upload.yaml [hunk 2]
```diff
@@ -190,23 +204,23 @@ variables:
...
         else ((states(sensor_windspeed) | float(default=0)) * 0.6213) }}
   windgustmph: >
     {% set unit_of_measurement = state_attr(sensor_windgust, 'unit_of_measurement') %}
-    {{ sensor_windgust if sensor_windgust == '' or unit_of_measurement != 'km/h'
+    {{ (states(sensor_windgust)|float(default=0)) if sensor_windgust == '' or unit_of_measurement != 'km/h'
```
Comment (dvandonkelaar) [inline wunderground_upload.yaml:211]: Why is this change needed? The current code works, just like other sensors (for example humidity)

File: wunderground_upload.yaml [hunk 3]
```diff
@@ -190,23 +204,23 @@ variables:
...
         else ((states(sensor_windgust) | float(default=0)) * 0.6213) }}
   winddir: >
     {{ sensor_winddir if sensor_winddir == ''
         else (states(sensor_winddir) | float(default=0)) }}
   rainin: >
     {% set unit_of_measurement = state_attr(sensor_rain, 'unit_of_measurement') %}
-    {{ sensor_rain if sensor_rain == ''
-        else ((states(sensor_rain) | float(default=0)) / 25.4 if unit_of_measurement == 'mm' else 1) }}
+    {{ (states(sensor_rain)|float(default=0)) if sensor_rain == '' or unit_of_measurement == 'in'
+        else ((states(sensor_rain) | float(default=0)) / 25.4 if unit_of_measurement == 'mm' ) }}
```
Comment (dvandonkelaar) [inline wunderground_upload.yaml:219]: This works, but seems nothing better than before

File: wunderground_upload.yaml [hunk 4]
```diff
@@ -190,23 +204,23 @@ variables:
...
   dailyrainin: >
     {% set unit_of_measurement = state_attr(sensor_dailyrain, 'unit_of_measurement') %}
-    {{ sensor_dailyrain if sensor_dailyrain == ''
-        else ((states(sensor_dailyrain) | float(default=0)) / 25.4 if unit_of_measurement == 'mm' else 1) }}
+    {{ (states(sensor_dailyrain)|float(default=0))if sensor_dailyrain == '' or unit_of_measurement == 'in'
+        else ((states(sensor_dailyrain) | float(default=0)) / 25.4 if unit_of_measurement == 'mm' ) }}
```
Comment (dvandonkelaar) [inline wunderground_upload.yaml:223]: This works, but seems nothing better than before

File: wunderground_upload.yaml [hunk 5]
```diff
@@ -248,10 +268,12 @@ variables:
           ,['dailyrainin', dailyrainin]
           ,['indoortempf', indoortempf]
           ,['indoorhumidity', indoorhumidity]
+          ,['solarradiation', solarradiation]
+          ,['UV', UV]
           ]%}
       {% if i[1] != '' %}
         {% set data.sensors= data.sensors + ['{}={}'.format(i[0], (i[1] if i[0] == 'rtfreq'
-                                              else (i[1] | round(3)) | string))] %}
+                                              else (i[1] | round(3, default="notavailable")) | string))] %}
```
Comment (dvandonkelaar) [inline wunderground_upload.yaml:276]: Why do we need a `notavailable` default?

File: wunderground_upload.yaml [hunk 6]
```diff
@@ -265,15 +287,14 @@ variables:
...
-        data:
-          url: "https://weatherstation.wunderground.com/weatherstation/updateweatherstation.php?{{ payload }}"
+  - alias: "Check if station_id and station_key exist"  
+    condition: and
+    conditions:
+      - "{{ station_id != '' }}"
+      - "{{ station_key != '' }}"
+  - service: !input curl_command_service
+    data:
+      url: "https://weatherstation.wunderground.com/weatherstation/updateweatherstation.php?{{ payload }}"#    if: >
```
Comment (dvandonkelaar) [inline wunderground_upload.yaml:297]: Why do we need to change this? This works now.

**Review by dvandonkelaar — CHANGES_REQUESTED** [review]

File: wunderground_upload.yaml (hunk 2 above)
Comment (jschwerdlin) [reply wunderground_upload.yaml:211]: Sensors that aren't native km/h (i.e. mph) won't get cast to a float otherwise. That makes it impossible to graph

File: wunderground_upload.yaml (hunk 1 above)
Comment (jschwerdlin) [reply wunderground_upload.yaml:207]: Sensors that aren't native km/h (i.e. mph) won't get cast to a float otherwise. That makes it impossible to graph

**Review by jschwerdlin — COMMENTED** [review]
All questions responded to

File: wunderground_upload.yaml (hunk 1 above)
Comment (dvandonkelaar) [reply wunderground_upload.yaml:207]: I see the problem. Only this results in a value of 0 if the sensor is not provided, which you don't want to upload. This is fixed in commit 19f366d

File: wunderground_upload.yaml (hunk 2 above)
Comment (dvandonkelaar) [reply wunderground_upload.yaml:211]: I see the problem. Only this results in a value of 0 if the sensor is not provided, which you don't want to upload. This is fixed in commit 19f366d

File: wunderground_upload.yaml (hunk 3 above)
Comment (dvandonkelaar) [reply wunderground_upload.yaml:219]: This results in a value of 0 if the sensor is not provided, which you don't want to upload. My code was wrong to have a value of 1 when the sensor is provided. This is fixed in commit 9f637e2.

File: wunderground_upload.yaml (hunk 4 above)
Comment (dvandonkelaar) [reply wunderground_upload.yaml:223]: This results in a value of 0 if the sensor is not provided, which you don't want to upload. My code was wrong to have a value of 1 when the sensor is provided. This is fixed in commit 9f637e2.

File: wunderground_upload.yaml [hunk 7]
```diff
@@ -215,6 +229,12 @@ variables:
   indoorhumidity: >
     {{ sensor_indoorhumidity if sensor_indoorhumidity == ''
         else (states(sensor_indoorhumidity) | float(default=0)) }}
+  solarradiation: >
+    {{ sensor_solarradiation if sensor_solarradiation == ''
+        else (states(sensor_solarradiation) | float(default=0)) / 126.7 }}
```
Comment (dvandonkelaar) [inline wunderground_upload.yaml:234]: What is this multiply factor that is used here?
Can you provide me all the attributes for this sensor?
The needed `unit_of_measurement` is `W/m²` according to [the API documentation](https://support.weather.com/s/article/PWS-Upload-Protocol?language=en_US).

File: wunderground_upload.yaml (hunk 5 above)
Comment (dvandonkelaar) [reply wunderground_upload.yaml:276]: This blueprint is created after HA 2021.12.0 and does not have this issue.
Can you provide me a sample to check out?

File: wunderground_upload.yaml (hunk 6 above)
Comment (dvandonkelaar) [reply wunderground_upload.yaml:297]: This still works for me.
Can you provide me a sample to check out?

**Review by dvandonkelaar — CHANGES_REQUESTED** [review]

File: wunderground_upload.yaml (hunk 7 above)
Comment (jschwerdlin) [reply wunderground_upload.yaml:234]: Native sensor reports in lux, but as you state, WUnderground wants w/m2.  Conversion factor between the two is 126.7
https://ambientweather.com/faqs/question/view/id/1452/

**Comment by jschwerdlin** [comment]
Regarding the default.  It's only a problem if the conversion fails.  I believe what I saw was the warning present in an earlier version of HA indicating the impending change and put the default in to remove the warning.  Again, this is discussed here:
https://community.home-assistant.io/t/template-warning-float-got-invalid-input-unknown-when-rendering-template/353198

How can I provide you sample data to try?  However, what I'm going to try to do is take your latest blueprint YAML file and see if it works for me as is now (for whatever reason).

File: wunderground_upload.yaml (hunk 7 above)
Comment (dvandonkelaar) [reply wunderground_upload.yaml:234]: Clear. Although I would like to check for the `unit_of_measurement` before doing the calculation.

**Comment by dvandonkelaar** [comment]
> How can I provide you sample data to try?
Please supply the trace of the automation run when it is failing. I made variables for the different properties and calculations, so probably I can see what is wrong.

**Review by dvandonkelaar — APPROVED** [review]

**Comment by dvandonkelaar** [comment]
This PR looks good and seems to work. Please create a new issue or PR if there are still some errors
